### PR TITLE
WIP Serverstats

### PR DESCRIPTION
**Pull request by aherlihy**
This change is [Reviewable](https://reviewable.io/reviews/mongodb-js/data-service/25)

### CODE REVIEW

File: lib/native-client.js [hunk 1]
```diff
@@ -106,6 +106,56 @@ class NativeClient {
...
 
   /**
+   * Get the currentOp.
+   *
+   * @param {Integer} includeAll - if true also list currently idle operations in the result.
+   * @param {Function} callback - The callback.
+   */
+  currentOp(includeAll, callback) { // TODO: deal with db versions? && do we want operations arg?
+
+    this.database.admin().command({'currentOp': 1, '$all': includeAll}, (error, result) => {
```
Comment (aherlihy) [inline lib/native-client.js]: I tried breaking up the lines but got a bunch of complains from the githook, is there a style guide?

Comment (durran) [inline lib/native-client.js]: We don't have a normal style guide that I know of but as you found out, eslint will complain if you violate the rules we set up, and runs always in the travis build and has a pre commit hook.

File: lib/native-client.js [hunk 2]
```diff
@@ -106,6 +106,56 @@ class NativeClient {
   }
 
   /**
+   * Get the currentOp.
+   *
+   * @param {Integer} includeAll - if true also list currently idle operations in the result.
+   * @param {Function} callback - The callback.
+   */
+  currentOp(includeAll, callback) { // TODO: deal with db versions? && do we want operations arg?
```
Comment (durran) [inline lib/native-client.js]: I think the ability to simply pass `true` for everything as you have implemented is fine.

File: lib/data-service.js [hunk 3]
```diff
@@ -286,6 +286,37 @@ class DataService extends EventEmitter {
...
 
   /**
+   * Returns the results of currentOp.
+   *
+   * @param {Object} operations - Document containing the operations to return information about.
+   * @param {Function} callback - The callback.
+   */
+  currentOp(operations, callback) {
+    debug(`#currentOp: ${operations}`);
+    this.client.currentOp(operations, callback);
```
Comment (durran) [inline lib/data-service.js]: Here you are passing an operations object but the `NativeClient` is expecting a `Boolean`. I would change `operations` to `includeAll` as you have in the native client.

File: lib/native-client.js (hunk 2 above)
Comment (aherlihy) [inline lib/native-client.js]: Done (removed TODO)

File: lib/data-service.js (hunk 3 above)
Comment (aherlihy) [inline lib/data-service.js]: Done.

**Pull request merged**